### ISSUE TITLE
IndexOutOfBoundsException During LoadTesting

### ISSUE THREAD

**Issue by hartmut-pq** [micrometer-metrics/micrometer#100]
Hi everyone,

Version: `0.9.2.RELEASE`

while doing some load testing I came across following exception thrown by micrometer core:
```
java.lang.IndexOutOfBoundsException: Index: 180, Size: 1
    at java.util.concurrent.CopyOnWriteArrayList.add(CopyOnWriteArrayList.java:462)
    at io.micrometer.core.instrument.stats.quantile.GKQuantiles.insertInBetween(GKQuantiles.java:241)
    at io.micrometer.core.instrument.stats.quantile.GKQuantiles.insertItem(GKQuantiles.java:188)
    at io.micrometer.core.instrument.stats.quantile.GKQuantiles.observe(GKQuantiles.java:94)
    at io.micrometer.core.instrument.stats.quantile.WindowSketchQuantiles$Block.insert(WindowSketchQuantiles.java:348)
    at io.micrometer.core.instrument.stats.quantile.WindowSketchQuantiles.insertElement(WindowSketchQuantiles.java:193)
    at io.micrometer.core.instrument.stats.quantile.WindowSketchQuantiles.observe(WindowSketchQuantiles.java:85)
    at io.micrometer.core.instrument.prometheus.internal.CustomPrometheusSummary$Child.observe(CustomPrometheusSummary.java:130)
    at io.micrometer.core.instrument.prometheus.PrometheusTimer.record(PrometheusTimer.java:42)
...
```

Not sure what my question is :-) I mainly wanted to flag it here for someone to have a look at...

**Comment by jkschneider** [comment]
Thanks for the report @hartmut-pq. We'll fix this for 0.11.0

**Comment by jkschneider** [comment]
As a workaround you might try CKMS quantiles for now.

**Comment by hartmut-pq** [comment]
Hi @jkschneider thanks for looking into and adding to the next milestone.
Is there a rough timeline / roadmap when 0.11.0 is to be released?

**Comment by jkschneider** [comment]
Later today or Monday at the latest.

**Comment by jkschneider** [comment]
Can't seem to reproduce this on the latest head. I've been trying to use the micrometer load test [here](https://github.com/micrometer-metrics/micrometer-load-test/blob/master/src/test/scala/io/micrometer/spring/perf/WebMvcApiSpec.scala#L12) after adding a quantile to the `io.micrometer.spring.samples.components.PersonController` sample controller. Any light you can shed on how your setup is different?

**Comment by jkschneider** [comment]
Fixed in 1.0.0-rc.3 when Histogram/Quantile implementations were replaced by HdrHistogram.
